### DISCUSSION TITLE
How to send POST request in Nextjs, Axios or something else?

### DISCUSSION THREAD

**Discussion by ivanjeremic** [vercel/next.js#13667] [Help] [answered]
```
axios
    .post('/https://extern-api.com/login', {
        username: "user",
        password: "123456"
    }
    )
    .then(res => {
      console.log('res', res.data);
    })
    .catch(err => {
      console.log('error in request', err);
    });
```

I tried to send a post request by calling this as function onClick & also inside /api routes in as an endpoint that sends a request for login. Both are not working so how we send POST requests in Nextjs ?

**Comment by finleygn** [comment, marked as answer]
This should just work, it could be the `/` before `https` if that's how it is in your code. If not an error log would be helpful

**Reply by ivanjeremic** [reply]
it's only a typo in this post. :(

**Reply by ivanjeremic** [reply]
It is working now as onClick, somehow axios inside /api routes is making problems no idea why but I mark this as the answer, Thanks.

**Reply by AfnanShawawreh** [reply]
can you tell me how you solve this problem?
and what you write in the API file?

**Comment by Allan-Nava** [comment]
it works?
